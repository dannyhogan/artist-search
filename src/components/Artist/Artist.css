@import url('https://fonts.googleapis.com/css?family=Montserrat&display=swap');

.Artist {
  h1 {
    font-family: 'Montserrat';
    color: black;
    font-weight: bolder;
  }
}
