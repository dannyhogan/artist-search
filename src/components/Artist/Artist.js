import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Releases from '../Releases/Releases';
import styles from './Artist.css';
import { getArtistReleases } from '../../services/artistApi';

export default class Artist extends Component {

  state = {
    releases: []
  }

  static propTypes = {
    match: PropTypes.object
  };


  componentDidMount() {
    getArtistReleases(this.props.match.params.id)
      .then(res => {
        this.setState({ releases: res.releases });
      });
  }

  render() {
    const { name } = this.props.match.params;
    return (
      <section className={styles.Artist}>
        <h1>Releases by {name}</h1>
        <Releases artist={name} releases={this.state.releases}/>
      </section>
    );
  }
}
