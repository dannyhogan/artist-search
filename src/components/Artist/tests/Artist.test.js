import React from 'react';
import { shallow } from 'enzyme';
import Artist from '../Artist';

describe('Artist component', () => {

  beforeEach(function() {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({ ok: true }));
  });


  it('renders Artist component', () => {
    const wrapper = shallow(<Artist match={{ params: 'Drake' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
