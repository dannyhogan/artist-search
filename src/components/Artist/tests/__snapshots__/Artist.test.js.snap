// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist component renders App 1`] = `
<section
  className="Artist"
>
  <h1>
    Releases by 
  </h1>
  <Releases
    releases={Array []}
  />
</section>
`;

exports[`Artist component renders Artist component 1`] = `
<section
  className="Artist"
>
  <h1>
    Releases by 
  </h1>
  <Releases
    releases={Array []}
  />
</section>
`;
